authority-portal-frontend/src/app/pages/catalog-page/catalog-page/catalog-page.component.ts

```diff
@@ -156,9 +163,40 @@ export class CatalogPageComponent implements OnInit, OnDestroy {
   }
 
   onDataOfferClick(dataOffer: CatalogDataOffer) {
+    this.openDataOfferDialog(dataOffer.assetId, dataOffer.connectorId);
+  }
+
+  private openDataOfferDialog(assetId: string, connectorId: string) {
     this.assetDetailDialogService
-      .open(dataOffer.assetId, dataOffer.connectorId, this.ngOnDestroy$)
-      .subscribe();
+      .open(assetId, connectorId, this.ngOnDestroy$)
+      .subscribe(
+        () => {
+          this.changeUrlToCatalogRoot();
+        },
+        () => {
+          this.changeUrlToCatalogRoot();
+        },
+      );
+    this.router.navigate([connectorId, assetId], {
+      relativeTo: this.route,
+      queryParams: {
+        environmentId: this.route.snapshot.queryParams.environmentId,
+      },
+    });
+    // BreadcrumbService builds the name from the URL which is nonsensical in casse of asset IDs
+    document.title = 'MDS Catalogue - Data Offer';
```

Comment:
typo, also, let's not maybe do that?

unless we switch it back well again

Reply:
It changes back automatically since it's always synchronized with the URL. I think this workaround here helps because asset IDs tend to generate titles like this
<img width="225" alt="image" src="https://github.com/user-attachments/assets/4cce2d55-6c65-4ade-bde4-dfd23fad7d39">